### ISSUE TITLE
Feedback on invalid CLI use

### ISSUE THREAD

**Issue by paulirish** [GoogleChrome/lighthouse-ci#297]
So I was exploring a bit on WSL today but managed to fatfinger a few invocations of LHCI while trying to use it without carefully reading the docs. I failed a few times. :) 

Here's a few things I attempted and what I saw:
```bash
# these two just exit quickly saying "Done running Lighthouse!"
lhci collect http://example.com  # no --url flag in front of the url
lhci collect --uautorun http://example.com # just a typo

# collects, but doesnt autorun
lhci collect --autorun --url=http://example.com 

# errors talking about staticDistDir but the real error is 'url' should be 'collect.url'
lhci autorun --url=http://example.com 

# collects and asserts, but then errors with `Error: Must provide token for LHCI target`
lhci autorun --collect.url=http://example.com 
```

yargs `strict()` mode solves the collect ones... so I thought it'd be good, but it breaks weirdly with `autorun` it seems. So I'm not sure the solution here, but I do think we should provide a louder failure.

**Comment by patrickhulce** [comment]
> but it breaks weirdly with autorun it seems

Yeah yargs strict mode is going to have a really bad time with the way we are so flexible on CLI flags. Everything relies on loose matching and passing down of flags to the layer below it.

I think the solution here is having the commands themselves check for invalid or missing properties when they're run. That's the only time at which all the configuration is actually present. We do this in a few places with a few select flags, but something more comprehensive (ideally with typo suggestion?) would be nice.
